### DISCUSSION TITLE
Help an idiot!

### DISCUSSION THREAD

**Discussion by Nopock** [PrismarineJS/mineflayer#2361] [Q&A]
const message15 = (`/gc [${interaction.user.username}] ${interaction.options.getString('content')}`);
bot.chat(message15);
is my thing and I get the error 
TypeError: bot.chat is not a function

**Comment by amoraschi** [comment]
Try doing it inside a "spawn" event

```js
bot.once("spawn", () => {
  // ...
})
```

**Reply by Nopock** [reply]
I have it inside a discord bot command

**Reply by amoraschi** [reply]
Well, you should check if the bot has joined the server, and then send that command

**Reply by Nopock** [reply]
Got a question (I found my code on a forum)

const bot = mineflayer.createBot({

  host: 'mc.hypixel.net', // Bot connects to the Hypixel Network
  username: process.env.EMAIL, // Your bot account email (preferably in a .env)
  password: process.env.PASSWORD // Your bot account password (preferably in a .env)
  
})

shouldn't this sign it in or whats the line of code to sign it in

**Reply by amoraschi** [reply]
Yes, what's the problem with it. Also, it's easier if you ask these questions in Discord, you'll get faster replies

Btw, to read .env files in NodeJS, you have to use the "dotenv" package

**Reply by Nopock** [reply]
Whats the discord and I use .env

**Reply by amoraschi** [reply]
First thing in the README.md:
https://discord.gg/GsEFRM8

And yes, I know you use .env, what I'm saying is this:
https://nodejs.dev/learn/how-to-read-environment-variables-from-nodejs

**Comment by xMdb** [comment]
slightly off topic, but if you are trying to do a guild chat bot, check out how I do it here: https://github.com/xMdb/hypixel-guild-chat-bot
